plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java

```diff
@@ -0,0 +1,167 @@
+package org.jenkinsci.plugins.structs.describable;
+
+import hudson.model.Describable;
+import org.jenkinsci.Symbol;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+/**
+ * A reflection-like view of a particular {@link Describable} instance.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class UninstantiatedDescribable implements Serializable {
+    private String symbol;
+    private String klass;
+    private final Map<String,Object> arguments;
+
+    public UninstantiatedDescribable(String symbol, String klass, Map<String, Object> arguments) {
+        this.symbol = symbol;
+        this.klass = klass;
+        this.arguments = arguments;
+    }
+
+    public UninstantiatedDescribable(Map<String, Object> arguments) {
```

Comment:
This seems really unsafe. IMO there should be one constructor taking a `@Nonnull String symbol`, another taking a `@Nonnull String klass`, and there should no setters for either.